### DISCUSSION TITLE
Does UniFi OS execute boot script before or after making initial WAN connections?

### DISCUSSION THREAD

**Discussion by ghost** [unifi-utilities/unifi-common#391] [Q&A] [answered]
Does UDM boot script apply custom scripts before or after UDM tries to make WAN connections after a reboot? I need to be certain UDM doesn't send any WAN traffic during reboot stage.

**Comment by boostchicken** [comment, marked as answer]
Well that depends,  if you are using eap_proxy or wpa it will be after.  It  runs at S95 in the main OS then comes up when networking is online in the unifi-os.   All this is to say I am not sure how deterministic it is.  I would say scripts most of the time run before your internet is up.  You can check in your scripts quite easily
